src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,76 @@
+package duke.task;
+
+/**
+ * Abstract class for a task.
+ *
+ * @author dexter-sim
+ * @version 0.1
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Creates a task with specified description.
+     *
+     * @param description The description of the task to be created.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Creates a task with specified description and completeness.
+     *
+     * @param description The description of the task to be created.
+     * @param isDone If the task is completed.
+     */
+    public Task(String description, boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    /**
+     * Returns a boolean if the description contains specified keyword.
+     *
+     * @param keyword A string to be checked.
+     * @return If description contains specified keyword.
+     */
+    public boolean match(String keyword) {
```

Comment:
Maybe it would be good to named Boolean methods to sound like booleans? Such as canMatch() or isMatch().